src/usdb_syncer/json.py

```diff
@@ -0,0 +1,79 @@
+"""Generates a JSON from the passed song list."""
+
+import datetime
+import json
+from typing import Any, Iterable
+
+from usdb_syncer import SongId
+from usdb_syncer.constants import VERSION
+from usdb_syncer.logger import get_logger
+from usdb_syncer.resource_dl import _url_from_resource
+from usdb_syncer.song_txt.headers import Headers
+from usdb_syncer.usdb_scraper import UsdbSong
+
+_logger = get_logger(__file__)
+
+MAX_SIZE = (100, 100)
+
+
+def get_headers(txt_path: str) -> Headers:
+    with open(txt_path, "r", encoding="utf-8") as file:
+        lines = [line for line in file.read().splitlines() if line]
+        return Headers.parse(lines, _logger)
+
+
+def generate_song_json(songs: Iterable[SongId], path: str) -> int:
+    date = datetime.datetime.now()
+    song_list: list[dict[str, Any]] = [
+        {
+            "id": song.song_id,
+            "artist": headers.artist,
+            "title": headers.title,
+            "year": int(headers.year) if headers.year else None,
+            "edition": (
+                None
+                if (not headers.edition or headers.edition == "None")
+                else headers.edition
+            ),
+            "genre": (
+                None
+                if (not headers.genre or headers.genre == "None")
+                else headers.genre
+            ),
+            "language": song.language,
+            "golden_notes": bool(song.golden_notes),  # force true/false instead of 1/0
+            "cover_url": (
+                meta.meta_tags.cover.source_url(_logger)
+                if meta.meta_tags.cover
+                else None
+            ),
+            "cover_meta": (
+                meta.meta_tags.cover.to_str("co") if meta.meta_tags.cover else None
+            ),
+            "audio_url": (
+                _url_from_resource(meta.meta_tags.audio)
+                if meta.meta_tags.audio
+                else None
+            ),
+            "video_url": (
+                _url_from_resource(meta.meta_tags.video)
+                if meta.meta_tags.video
+                else None
+            ),
+            "duet": (
+                meta.meta_tags.player1 is not None
+                and meta.meta_tags.player2 is not None
+            ),
+        }
+        for song_id in songs
+        if (song := UsdbSong.get(song_id))
+        and (meta := song.sync_meta)
+        and (txt := meta.txt)
+        and (headers := get_headers(str(meta.path.with_name(txt.fname))))
+    ]
+    content = {"songs": song_list, "date": str(date), "syncer_version": VERSION}
```

Comment:
The format should have its own versioning.

Reply:
I thought some information might be related on a syncer update and therefore written differently. This would lead to different output of the same song even though the actual json format did not change. Also I thought the JSON format will change less frequently than the syncer version so in case some compatibility issues arise we could mark ranges of syncer versions as compatible. Introducing an additional versioning seemed confusing to me.

Reply:
> I thought some information might be related on a syncer update and therefore written differently. [...]  we could mark ranges of syncer versions as compatible.

That seems to defeat the whole purpose of a format. Consumers would need to be aware of the producer and all its versions.

Reply:
I see, so I introduce a json export format version as local variable independent from the syncer version and start with `0.1.0`?

Reply:
Yes, I would remove `syncer_version`, call the field `version` and make it a simple integer.

Reply:
added `export_version` in ce58e2c

Reply:
just a single integer `version` in b46a14a